app/src/main/java/com/github/mik629/android/fundamentals/App.kt

```diff
@@ -0,0 +1,28 @@
+package com.github.mik629.android.fundamentals
+
+import android.app.Application
+import com.github.mik629.android.fundamentals.di.AppComponent
+import com.github.mik629.android.fundamentals.di.DaggerAppComponent
+import timber.log.Timber
+import javax.inject.Inject
+
+class App : Application() {
+
+    @Inject
+    lateinit var tree: Timber.Tree
+
+    override fun onCreate() {
+        super.onCreate()
+
+        appComponent = DaggerAppComponent.builder()
+            .appContext(this)
+            .apiUrl(BuildConfig.BASE_URL)
+            .build()
+        appComponent.inject(this)
+        Timber.plant(tree)
+    }
+
+    companion object {
```

Comment:
avoid using companion for dagger components use just as class property
an as it used in onCreate as appComponent.inject(this)
u can use by lazy{} for creating component

Reply:
Here goes the [issue](https://github.com/mik629/AndroidFundamentalsApp/issues/12)